### PR TITLE
[2201.3.0-stage] Refactor workspace service to gracefully access documents of generated folder with didOpen event

### LINKED ISSUE #38234: [Bug]: [Project API] `document` API is throwing an exception for generated IDL client
### Description

I generated an IDL client and reloaded the VSCode window. When I was trying to make a change in the generated client the following exception was thrown.

image

https://github.com/ballerina-platform/ballerina-lang/blob/656430fde21a51e5c01193a72127baf670bc4efe/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/workspace/BallerinaWorkspaceManager.java#L1108

Here we are trying to invoke the `document` method and get the document for the path of the generated idl client. It throws an exception. Due to this the project can't be compiled. 

### Steps to Reproduce

Mentioned above

### Affected Version(s)

2201.3.0-stage

### OS, DB, other environment details and versions

_No response_

### Related area

-> Editor

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by malinthar**
## Purpose
$subject

Fixes #38234

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionRouter.java [hunk 1]
```diff
@@ -65,10 +66,15 @@ public static List<CodeAction> getAvailableCodeActions(CodeActionContext ctx) {
                 = CodeActionProvidersHolder.getInstance(ctx.languageServercontext());
 
         // Get available range based code-actions
-        SyntaxTree syntaxTree = ctx.currentSyntaxTree().orElseThrow();
+        Optional<SyntaxTree> syntaxTree = ctx.currentSyntaxTree();
+        if (syntaxTree.isEmpty()) {
+            clientLogger.logTrace(LSContextOperation.TXT_CODE_ACTION.getName() + " " +
+                    " Syntax tree is empty for file" + ctx.fileUri());
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionRouter.java:72]:
```suggestion
                    " Syntax tree is empty for file " + ctx.fileUri());
```

**Review by IMS94 — CHANGES_REQUESTED** [review]
LGTM! There's a minor suggestion.

**Review by IMS94 — APPROVED** [review]

**Comment by IMS94** [comment]
@malinthar we should send this to master as well right?

**Review by gayalkuruppu — APPROVED** [review]
LGTM

**Pull request merged**